libcudacxx/codegen/generators/definitions.h

```diff
@@ -0,0 +1,192 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef DEFINITIONS_H
+#define DEFINITIONS_H
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include <fmt/format.h>
+
+enum class Mmio
+{
+  Disabled,
+  Enabled,
+};
+
+static std::string mmio(Mmio m)
+{
+  static std::map mmio_map{
+    std::pair{Mmio::Disabled, ""},
+    std::pair{Mmio::Enabled, ".mmio"},
+  };
+  return mmio_map[m];
```

Comment:
Suggestion: I guess we don't care about runtime and memory consumption here. Otherwise this could just be a plain array:
```suggestion
  static const char* mmio_map[]{
    "",
    ".mmio",
  };
  return mmio_map[std::to_underlying(m)];
```